--- capiba_v2.pine
+++ capiba_v2.pine
@@ -370,7 +370,7 @@
                 eachFairValueGap.topBox.set_border_color(color.new(color.gray, 80))
                 eachFairValueGap.bottomBox.set_border_color(color.new(color.gray, 80))
                 eachFairValueGap.mitigated := true
-            else if fullyMitigatedBull or fullyMitigatedBear
+            if fullyMitigatedBull or fullyMitigatedBear
                 eachFairValueGap.topBox.delete()
                 eachFairValueGap.bottomBox.delete()
                 if not na(eachFairValueGap.midLine)
